src/query/builder.cr

```diff
@@ -0,0 +1,109 @@
+# Data structure which will allow chaining of query components,
+# nesting of boolean logic, etc.
+#
+# Should return self, or another instance of Builder wherever
+# chaining should be possible.
+#
+# Current query syntax:
+# - where(field: value) => "WHERE field = 'value'"
+#
+# Hopefully soon:
+# - Model.where(field: value).not( Model.where(field2: value2) )
+# or
+# - Model.where(field: value).not { where(field2: value2) }
+#
+# - Model.where(field: value).or( Model.where(field3: value3) )
+# or
+# - Model.where(field: value).or { whehre(field3: value3) }
+class Query::Builder(T)
+  alias FieldName = String
+  alias FieldData = DB::Any
+
+  def initialize(@boolean_operator = :and)
+    @fields = {} of FieldName => FieldData
+    @order  = [] of NamedTuple(field: String, direction: String)
+  end
+
+  def compile
+    Compiled(T).new self
+  end
+
+  def runner
+    Runner(T).new compile
+  end
+
+  def where(**matches)
+    matches.each do |field, data|
+      @fields[field.to_s] = data
+    end
+
+    self
+  end
+
+  def order(field : Symbol)
```

Comment:
what about using a tuple instead `order(updated_at: :asc)`?

Reply:
The other overload does exactly that. This method is meant to be called: `.order(:id)` and assumes `:asc`.